src/data_pipeline_api/file_api.py

```diff
@@ -0,0 +1,177 @@
+from io import IOBase
+from datetime import datetime
+from pathlib import Path
+from typing import Union
+from functools import wraps
+from hashlib import sha1
+import yaml
+
+from data_pipeline_api.metadata import Metadata, MetadataKey
+from data_pipeline_api.metadata_store import MetadataStore
+from data_pipeline_api.overrides import Overrides
+
+
+class FileAPI:
+    def __init__(
+        self, config_filename: Union[Path, str],
+    ):
+        config_filename = Path(config_filename)
+
+        # Compute a run_id.
+        with open(config_filename, "rb") as file:
+            m = sha1(file.read())
+        m.update(str(datetime.now()).encode())
+        self.run_id = m.hexdigest()
+
+        try:
+            with open(config_filename) as config_file:
+                config = yaml.safe_load(config_file)
```

Comment:
just to check my understanding - the safe loading is OK because we're assuming everything is going to be dicts/lists/strs and nothing more complex?

Reply:
That is my understanding as well 🤷